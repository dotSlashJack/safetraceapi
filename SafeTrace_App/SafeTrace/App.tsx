import React, { useState, useEffect, useRef } from "react";
import { Platform, StatusBar } from "react-native";
import { SplashScreen } from "expo";
import * as Font from "expo-font";
import { Ionicons } from "@expo/vector-icons";
import { ThemeProvider } from "styled-components/native";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";

import { BaseLayout } from "./styles";
import useLinking from "./navigation/useLinking";
import RootNavigator from "./navigation/RootNavigator";
import { Theme } from "./Theme";
import { configureStore } from "./store";

interface IProps {
  skipLoadingScreen: any;
}

export default function App(props: IProps) {
  const [isLoadingComplete, setLoadingComplete] = useState(false);
  const [initialNavigationState, setInitialNavigationState] = React.useState();
  const containerRef = useRef();
  const { getInitialState } = useLinking(containerRef);
  const storeRef = configureStore();

  const StatusBarNode = Platform.OS === "ios" && (
    <StatusBar barStyle="default" />
  );

  useEffect(() => {
    const loadResourcesAndDataAsync = async () => {
      try {
        SplashScreen.preventAutoHide();

        // Load our initial navigation state
        setInitialNavigationState((await getInitialState()) as any);

        // Load fonts
        await Font.loadAsync({
          ...Ionicons.font,
          "space-mono": require("./assets/fonts/SpaceMono-Regular.ttf"),
        });
      } catch (e) {
        // We might want to provide this error information to an error reporting service
        console.warn(e);
      } finally {
        setLoadingComplete(true);
        SplashScreen.hide();
      }
    };

    loadResourcesAndDataAsync();
  }, []);

  if (!isLoadingComplete && !props.skipLoadingScreen) {
    return <BaseLayout></BaseLayout>;
  } else {
    return (
      <BaseLayout>
        {StatusBarNode}
        <ThemeProvider theme={Theme}>
          <Provider store={storeRef.store}>
            <PersistGate loading={null} persistor={storeRef.persistor}>
              <RootNavigator
                containerRef={containerRef}
                initialState={initialNavigationState}
              />
            </PersistGate>
          </Provider>
        </ThemeProvider>
      </BaseLayout>
    );
  }
}
