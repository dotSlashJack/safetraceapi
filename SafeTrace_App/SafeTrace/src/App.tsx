import React, { useState, useEffect, useRef } from "react";
import { Platform, StatusBar } from "react-native";
import { ThemeProvider } from "styled-components/native";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { NavigationContainer } from "@react-navigation/native";

import { BaseLayout } from "../styles";
import useLinking from "./navigation/useLinking";
import RootNavigator from "./navigation/RootNavigator";
import { Theme } from "../Theme";
import { configureStore } from "./store";

interface IProps {
  skipLoadingScreen?: any;
}

export default (props: IProps) => {
  const [isLoadingComplete, setLoadingComplete] = useState(false);
  const [initialNavigationState, setInitialNavigationState] = React.useState();
  const containerRef = useRef();
  const { getInitialState } = useLinking(containerRef);
  const storeRef = configureStore();

  const StatusBarNode = Platform.OS === "ios" && (
    <StatusBar barStyle="default" />
  );

  useEffect(() => {
    const loadResourcesAndDataAsync = async () => {
      try {
        // Load our initial navigation state
        setInitialNavigationState((await getInitialState()) as any);
      } catch (e) {
        // We might want to provide this error information to an error reporting service
        console.info(e);
      } finally {
        setLoadingComplete(true);
      }
    };

    loadResourcesAndDataAsync();
  }, []);

  if (!isLoadingComplete && !props.skipLoadingScreen) {
    return <BaseLayout />;
  } else {
    return (
      <BaseLayout>
        {StatusBarNode}
        <ThemeProvider theme={Theme}>
          <Provider store={storeRef.store}>
            <PersistGate loading={null} persistor={storeRef.persistor}>
              <RootNavigator
                containerRef={containerRef}
                initialState={initialNavigationState}
              />
            </PersistGate>
          </Provider>
        </ThemeProvider>
      </BaseLayout>
    );
  }
};
