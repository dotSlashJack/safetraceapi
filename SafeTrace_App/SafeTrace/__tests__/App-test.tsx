import React from 'react';
// import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
  Linking: {
    makeUrl: jest.fn(),
  },
  SplashScreen: {
    preventAutoHide: jest.fn(),
  }
}));

// jest.mock('../navigation/AppNavigator', () => 'AppNavigator');

describe('App', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    // NavigationTestUtils.resetInternalState();
  });

  it('renders correctly', () => {
    renderer.create(<App />);
  })

  // TODO: Fix test
  // it('has 1 child', () => {
  //   const tree = renderer.create(<App />).toJSON();
  //   // expect(tree.children.length).toBe(1);
  // })

  it(`renders the loading screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: Fix test
  // it(`renders the root without loading screen`, () => {
  //   const tree = renderer.create(<App skipLoadingScreen={true} />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
