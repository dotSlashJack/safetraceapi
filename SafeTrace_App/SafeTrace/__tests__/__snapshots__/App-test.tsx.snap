// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
